test/SortTest.cpp

```diff
@@ -172,3 +173,32 @@ TEST(Sort, SimpleMemberFunctions) {
     EXPECT_EQ(84.0, s.getMultiplicity(1));
   }
 }
+
+TEST(Sort, verifyOperationIsPreemptivelyAbortedWithNoRemainingTime) {
+  VectorTable input;
+  // Make sure the estimator estimates a couple of ms to sort this
+  for (int64_t i = 0; i < 1000; i++) {
+    input.push_back({0, i});
+  }
+  auto inputTable = makeIdTableFromVector(input, &Id::makeFromInt);
+  Sort sort = makeSort(std::move(inputTable), {1, 0});
+  const_cast<SortPerformanceEstimator&>(
+      sort.getExecutionContext()->getSortPerformanceEstimator())
+      .computeEstimatesExpensively(
+          ad_utility::makeUnlimitedAllocator<ValueId>(),
+          std::numeric_limits<size_t>::max());
+
+  sort.recursivelySetTimeConstraint(0ms);
+
+  EXPECT_THROW(
+      {
+        try {
+          sort.getResult(true);
+        } catch (const ad_utility::AbortException& exception) {
+          EXPECT_THAT(exception.what(),
+                      ::testing::HasSubstr("time estimate exceeded"));
+          throw;
+        }
+      },
+      ad_utility::AbortException);
```

Comment:
This is a combination between `EXPECT_THROW` and `AD_EXPECT_THROW_WITH_MESSAGE` (in `GTestHelpers.h`).
Factor this out as a combined check `AD_EXPECT_THROW_WITH_MESSAGE_AND_TYPE(sort.getResult(true), HasSubstr(...), AbortException)`